src/awkward/operations/ak_drop_none.py

```diff
@@ -0,0 +1,82 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import awkward as ak
+
+np = ak.nplikes.NumpyMetadata.instance()
+
+
+def drop_none(array, axis=None, highlevel=True, behavior=None):
```

Comment:
I hadn't been thinking of this, but I see the argument for making `axis=None` be the default. I think you're right in assuming that most users would want to eliminate _all_ missing values, regardless of depth. Dropping missing values at a particular level of depth is more niche, but also necessary.